### DISCUSSION TITLE
How can I set cookies correctly

### DISCUSSION THREAD

**Discussion by liming-xxw** [nuxt/nuxt#17798] [Questions] [answered]
I use usecookies
![image](https://user-images.githubusercontent.com/56504602/159442921-ba6c0fab-d002-47c3-9821-59deecd3696c.png)
But it will prompt an error ，  ReferenceError: useCookie is not defined 
![image](https://user-images.githubusercontent.com/56504602/159443143-9f5b2cc4-f4b3-4ea8-a1de-7525148569b6.png)
If you can, ask for the correct usage ，thank you ,谢谢

**Comment by Diizzayy** [comment, marked as answer]
@liming-xxw The code below should meet your needs

```ts
<script lang="ts" setup>
type IUser = { name: string };
const user = useCookie<IUser>("user", { default: () => ({ name: "example" }) });

const setUser = (name: string) => {
  user.value = { name };
};
</script>
```

If the user cookie isn't present on the client, it will default to `{ name: "example" }`

to update the cookie, you **must** reassign `user.value`, so for example, given the code above, setting `user.value.name = name` would in fact update the cookie variable but it wouldn't set the cookie.

**Reply by liming-xxw** [reply]
thank you,
